.github/workflows/pr.yaml

```diff
@@ -71,6 +86,8 @@ jobs:
       - uses: taiki-e/install-action@v2
         with:
           tool: nextest@0.9.84
+      - name: Rust Cache
+        uses: Swatinem/rust-cache@v2
```

Comment:
It looks like this speeds up overall builds by 1m. Not sure we want to add a dependency to our build pipeline with caching (which notoriously can cause some weird build issues). A bit nitpicky, but I'm always anxious about complicating the build in ways that may not be reproducible locally.

cc @criccomini to get your thoughts on whether you think this is worth it.

<img width="844" alt="image" src="https://github.com/user-attachments/assets/b1320420-eb31-429d-9ca5-3650c093ed40" />
<img width="839" alt="image" src="https://github.com/user-attachments/assets/0957d015-9475-46a5-ad50-5110ca50fd4c" />